cadence/src/sinks/unix.rs

```diff
@@ -202,8 +226,13 @@ impl BufferedUnixMetricSink {
     where
         P: AsRef<Path>,
     {
+        let uwa = UnixWriteAdapter::new(socket, path);
+        let bytes_sent = uwa.bytes_sent();
```

Comment:
nit: Instead of getting the counter back from `UnixWriteAdapter`, I'd rather we created the counter in this method and passed `Arc<AtomicU64>` into `UnixWriteAdapter::new()`.